templates/bbb.nomad

```diff
@@ -0,0 +1,217 @@
+        ]
+        port_map {
+          redis = 6379
+        }
+        labels {
+          liquid_task = "bbb-redis"
+        }
+      }
+
+      resources {
+        cpu = 500
+        memory = 512
+        network {
+          port "redis" {}
+          mbits = 1
+        }
+      }
+
+      service {
+        name = "bbb-redis"
+        port = "redis"
+        tags = ["fabio-:${config.port_bbb_redis} proto=tcp"]
+
+        check {
+          name = "bbb-redis-alive"
+          initial_status = "critical"
+          type = "tcp"
+          interval = "${check_interval}"
+          timeout = "${check_timeout}"
+        }
+        check_restart {
+          limit = 5
+          grace = "980s"
+        }
+      }
+    }
+
+    task "greenlight" {
+      ${ task_logs() }
+      leader = false
+
+      constraint {
+        attribute = "{% raw %}${meta.liquid_volumes}{% endraw %}"
+        operator = "is_set"
+      }
+
+      affinity {
+        attribute = "{% raw %}${meta.liquid_large_databases}{% endraw %}"
+        value     = "true"
+        weight    = 100
+      }
+
+      driver = "docker"
+      ${ shutdown_delay() }
+
+      template {
+        data = <<-EOF
+        {{- with secret "liquid/bbb/bbb.postgres" }}
+        DATABASE_URL = "postgres://greenlight:{{.Data.secret_key | toJSON }}@{{env "attr.unique.network.ip-address"}}:${config.port_bbb_pg}/greenlight_production"
+        {{- end }}
+        REDIS_URL = "redis://{{ env "attr.unique.network.ip-address"}}:${config.port_bbb_redis}"
+        BIGBLUEBUTTON_ENDPOINT = "${config.bbb_endpoint}"
+        BIGBLUEBUTTON_SECRET = "${config.bbb_secret}"
+        {{- with secret "liquid/bbb/bbb.secret_key_base" }}
+        SECRET_KEY_BASE = {{.Data.secret_key | toJSON }}
+        {{- end }}
+        RELATIVE_URL_ROOT = "/"
+        {{- with secret "liquid/bbb/auth.oauth2" }}
+        OPENID_CONNECT_CLIENT_ID = {{.Data.client_id | toJSON }}
+        OPENID_CONNECT_CLIENT_SECRET = {{.Data.client_secret | toJSON }}
+        {{- end }}
+        OPENID_CONNECT_ISSUER="https://${liquid_domain}"
+        OPENID_CONNECT_REDIRECT="https://bbb.${liquid_domain}/"
+        EOF
+        destination = "local/greenlight.env"
+        env = true
+      }
+
+      config {
+        image = "${config.image('bbb-gl')}"
+        volumes = [
+          "{% raw %}${meta.liquid_volumes}{% endraw %}/bbb/greenlight/data/:/usr/src/app/storage/",
+        ]
+        port_map {
+          bbb-gl = 3000
+        }
+        labels {
+          liquid_task = "bbb_greenlight"
+        }
+        memory_hard_limit = 2000
+      }
+
+      resources {
+        memory = 450
+        cpu = 150
+        network {
+          mbits = 1
+          port "bbb-gl" {}
+        }
+      }
+
+      service {
+        name = "bbb-gl"
+        port = "bbb-gl"
+        tags = [
+        "traefik.enable=true",
+        "traefik.frontend.rule=Host:bbb.${liquid_domain}",
```

Comment:
Putting "greenlight" directly on the Traefik means people will see (and be able to attack/bruteforce) the login form of that app directly.

We usually try to wrap all possible HTTP/S traffic in the "authproxy" microservice, even for services with already have OAuth2 integration - this ensures browsers have valid login before even connecting them to the app.

Some services (like Matrix) aren't behind that proxy because they break external app functionality - is this something that we are interested here too? (for example BBB on tablets/phones?)

Reply:
I didn't quite get that part, I think I'll look at dokuwiki template-authproxy, and arrange a test from scratch with that.

Reply:
I tried to mimic config based on what's done with dokuwiki, but it doesn't work as is, maybe we'll need to tweak some files, but in the meantime, I think we can go with the actual version and make a following issue regarding that part

Reply:
resolved with https://github.com/liquidinvestigations/node/pull/361